packages/vm/lib/runTx.ts

```diff
@@ -97,6 +98,23 @@ export default async function runTx(this: VM, opts: RunTxOpts): Promise<RunTxRes
   debug('-'.repeat(100))
   debug(`tx checkpoint`)
 
+  // Is it an Access List transaction?
+  if (opts.tx.transactionType == 1 && this._common.isActivatedEIP(2929)) {
```

Comment:
Wrong equality operator `==` (also in `runBlock`), this check is also not sufficient respectively not working atm since - currently - the `LegacyTransaction` is also returning type 1 and therefore the cast will be wrong (this is currently breaking the API tests along the `eip2929.ts` test file).

Reply:
LegacyTransaction should have type 0, weird that I messed this up.